lib/extensions.js

```diff
@@ -197,6 +197,10 @@ module.exports = function (realmConstructor) {
 
         // First configure any timeOut and corresponding behavior.
         let openPromises = [];
+
+        // This will be set to true by the first promise to resolve.  The next promise will use this to stop execution
+        let fallbackTimeOutId = null;
```

Comment:
```suggestion
        // Id of the timer triggering the timeout
        let timeoutId = null;
```

There are no other timeout ids.

Reply:
Good catch.  Fix is in